src/Service/MagicLinks/MagicLinkService.cs

```diff
@@ -0,0 +1,49 @@
+using Passwordless.Common.Services.Mail;
+using Passwordless.Service.MagicLinks.Models;
+using Passwordless.Service.Models;
+
+namespace Passwordless.Service.MagicLinks;
+
+public class MagicLinkService
+{
+    private readonly IFido2Service _fido2Service;
+    private readonly IMailProvider _mailProvider;
+
+    public MagicLinkService(IFido2Service fido2Service, IMailProvider mailProvider)
+    {
+        _fido2Service = fido2Service;
+        _mailProvider = mailProvider;
+    }
+
+    public async Task SendMagicLink(MagicLinkDTO dto)
+    {
+        var token = await _fido2Service.CreateSigninToken(new SigninTokenRequest(dto.UserId));
+
+        var link = new Uri(dto.LinkTemplate.Replace("<token>", token));
+
+        await _mailProvider.SendAsync(new MailMessage
```

Comment:
blocking: I think we need to introduce a overload on this interface - or we need to tweak the implementation to look at the messageStream property and use the correct API keys (or underlying provider)